.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,111 @@
+name: Benchmark
+on:
+  schedule:
+    - cron: "0 2 * * *"
+permissions:
+  contents: read
+jobs:
+  BENCHMARK:
+    name: BENCHMARK
+    if: github.repository == 'php/php-src'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install dependencies
+        run: |
+          set -ex
+          sudo apt-get update
+          sudo apt-get install gpg
+
+          wget -O- https://apt.releases.hashicorp.com/gpg | sudo gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
+          gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint
+          echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
+          sudo apt update
+          sudo apt install terraform=1.5.7-*
+      - name: Checkout benchmark suite
+        uses: actions/checkout@v4
+        with:
+          repository: 'kocsismate/php-version-benchmarks'
+          ref: 'main'
+          fetch-depth: 1
+          path: 'php-version-benchmarks'
+      - name: Checkout php-src
+        uses: actions/checkout@v4
+        with:
+          repository: 'php/php-src'
+          ref: 'master'
+          fetch-depth: 100
+          path: 'php-version-benchmarks/tmp/php_master'
+      - name: Copy-paste the same repo content for benchmarking JIT
+        run: |
+          set -e
+    
+          cp -r "php-version-benchmarks/tmp/php_master/" "php-version-benchmarks/tmp/php_master_jit"
+      - name: Setup benchmark config
+        run: |
+          set -e
+
+          cp ./php-version-benchmarks/config/infra/aws/x86_64-metal.ini.dist ./php-version-benchmarks/config/infra/aws/x86_64-metal.ini
+          ESCAPED_DOCKER_REGISTRY=$(printf '%s\n' "${{ secrets.PHP_VERSION_BENCHMARK_DOCKER_REGISTRY }}" | sed -e 's/[\/&]/\\&/g')
+          sed -i "s/INFRA_DOCKER_REGISTRY=public.ecr.aws\/abcdefgh/INFRA_DOCKER_REGISTRY=$ESCAPED_DOCKER_REGISTRY/g" ./php-version-benchmarks/config/infra/aws/x86_64-metal.ini
+
+          cp ./php-version-benchmarks/build/infrastructure/config/aws.tfvars.dist ./php-version-benchmarks/build/infrastructure/config/aws.tfvars
+          sed -i 's/access_key = ""/access_key = "${{ secrets.PHP_VERSION_BENCHMARK_AWS_ACCESS_KEY }}"/g' ./php-version-benchmarks/build/infrastructure/config/aws.tfvars
+          sed -i 's/secret_key = ""/secret_key = "${{ secrets.PHP_VERSION_BENCHMARK_AWS_SECRET_KEY }}"/g' ./php-version-benchmarks/build/infrastructure/config/aws.tfvars
+
+          cp ./php-version-benchmarks/config/php/master.ini.dist ./php-version-benchmarks/config/php/master1.ini
+          YESTERDAY="$(date -d "-2 day 13:00" '+%Y-%m-%d')"
```

Comment:
Intel's benchmarks years ago used a single baseline commit: https://externals.io/message/100532 They used a specific PHP 7.0 commit. I'm not sure what the best would be for us, but it may be useful to have a clear understanding about the performance of a PHP version since development started (e.g. the baseline could be the first commit after the latest minor PHP version was branched|)

Reply:
@arnaud-lb What do you think about this possibility? Should we add a specific commit (i.e. the first commit after PHP 8.4 is branched) to the benchmark? This way, we would compare the latest commit with the baseline and yesterday's commit.

Reply:
I've just added it, because I think it will come handy if we also have a single fixed commit against which we can compare changes.